### PR TITLE
add support for all DOM events to Marker

### PR DESCRIPTION
**Pull request by charkour**
Thank you for supporting this package!

When using the Marker component, I'd like to attach more event handlers in this capacity below. `onMouseEnter` is not currently supported by react-map-gl. The proposed code will add the proper event listeners when the props are provided. 

```tsx
      <Map
        mapStyle="mapbox://styles/mapbox/streets-v11"
        mapboxAccessToken={process.env.NEXT_PUBLIC_MAPBOX_TOKEN}
      >
        {Object.values(markers).map((marker) => (
          <Marker
            key={marker.id}
            onMouseEnter={() => {
              console.log("marker hovered", marker.id);
              setHighLightedId(marker.id);
            }}
            style={{
              backgroundColor:
                marker.id === highLightedId ? "red" : "blue",
            }}
            longitude={marker.lng}
            latitude={marker.lat}
          />
        ))}
      </Map>
```

If this is a helpful addition, please let me know! We can add tests and improve the documentation. 

Thank you

### CODE REVIEW

File: src/components/marker.ts [hunk 1]
```diff
@@ -102,13 +99,20 @@ function Marker(props: MarkerProps) {
...
-        target: mk,
-        originalEvent: e
-      });
-    });
+    for (const propName of Object.keys(thisRef.current.props)) {
+      const isHandler = /^on[A-Z]/.test(propName);
+      if (isHandler) {
+        // slice(2) to remove 'on' prefix
+        const eventName = propName.slice(2).toLowerCase();
+        mk.getElement().addEventListener(eventName, (e: MapboxEvent<React.SyntheticEvent>) => {
```
Comment (charkour) [inline src/components/marker.ts:107]: We'll have to see what the proper event typing is.

File: src/components/marker.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ import type {
 import {MapContext} from './map';
 import {arePointsEqual} from '../utils/deep-equal';
 
-export type MarkerProps = {
+export type MarkerProps = Omit<React.HTMLAttributes<HTMLElement>, 'color' | 'draggable'> & {
```
Comment (charkour) [inline src/components/marker.ts:20]: I was not sure if `color` and `draggable` have different types that the default `HTMLElement` type so I omitted them.

**Review by Pessimistress — COMMENTED** [review]
Thanks for the proposal. I have a couple of concerns about the current implementation:

- You are accepting all prop types from `React.HTMLAttributes<HTMLElement>` but only the `on*` ones seem to be used?
- Event handlers are attached at the initial mount, which means the user cannot add new callbacks after that. I can see that behavior potentially leads to confusion in some cases.

Does it make sense to use `createPortal` to render a `<div {...props}>` into the marker element? All prop handling will be deferred to React then.

**Comment by charkour** [comment]
Using a portal sounds great! (Sorry for the late reply)

**Comment by charkour** [comment]
I opted for using an event listener attached via a ref.

**Pull request closed without merging**